src/plugins/inputs/connect.js

```diff
@@ -5,21 +5,26 @@ const Msg = require("../../models/msg");
 exports.commands = ["connect", "server"];
 exports.allowDisconnected = true;
 
-exports.input = function({irc}, chan, cmd, args) {
+exports.input = function(network, chan, cmd, args) {
 	if (args.length === 0) {
-		if (!irc || !irc.connection) {
+		network.userDisconnected = false;
+		this.save();
+
+		const irc = network.irc;
+
+		if (!irc) {
 			return;
 		}
 
-		if (irc.connection.connected) {
+		if (irc.connection && irc.connection.connected) {
 			chan.pushMessage(this, new Msg({
 				type: Msg.Type.ERROR,
 				text: "You are already connected.",
 			}));
 			return;
 		}
 
-		irc.connection.connect();
```

Comment:
For some reason this incorrectly called `connect` on the connection, and not the client instance itself.